library/core/src/intrinsics.rs

```diff
@@ -2242,13 +2242,33 @@ pub const unsafe fn copy<T>(src: *const T, dst: *mut T, count: usize) {
 /// assert_eq!(*v, 42);
 /// ```
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_const_unstable(feature = "const_ptr_write", issue = "86302")]
 #[inline]
-pub unsafe fn write_bytes<T>(dst: *mut T, val: u8, count: usize) {
+pub const unsafe fn write_bytes<T>(dst: *mut T, val: u8, count: usize) {
     extern "rust-intrinsic" {
+        #[rustc_const_unstable(feature = "const_ptr_write", issue = "86302")]
         fn write_bytes<T>(dst: *mut T, val: u8, count: usize);
     }
 
-    debug_assert!(is_aligned_and_not_null(dst), "attempt to write to unaligned or null pointer");
+    #[cfg(not(bootstrap))]
+    const fn compiletime_check<T>(_ptr: *mut T) {
+        // The CTFE-engine checks if the pointer is correctly aligned and not null.
+    }
```

Comment:
similarly to the other PR, please just call this function `nop` and mention on the const_eval_select call that it's just a debug assertion and so it's not a problem if CTFE does not check it, even if it checks some of the rules